ecs-agent/introspection/server_test.go

```diff
@@ -0,0 +1,171 @@
+//go:build unit
+// +build unit
+
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package introspection
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"strconv"
+	"testing"
+	"time"
+
+	mock_v1 "github.com/aws/amazon-ecs-agent/ecs-agent/introspection/v1/mocks"
+	mock_metrics "github.com/aws/amazon-ecs-agent/ecs-agent/metrics/mocks"
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+var runtimeStatsConfigForTest = false
+
+func TestNewServerErrors(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	agentState := mock_v1.NewMockAgentState(ctrl)
+	metricsFactory := mock_metrics.NewMockEntryFactory(ctrl)
+	t.Run("state is required", func(t *testing.T) {
+		_, err := NewServer(nil, metricsFactory)
+		assert.EqualError(t, err, "state cannot be nil")
+	})
+	t.Run("metrics factory is required", func(t *testing.T) {
+		_, err := NewServer(agentState, nil)
+		assert.EqualError(t, err, "metrics factory cannot be nil")
+	})
+}
+
+func TestNewServerDefaults(t *testing.T) {
```

Comment:
nit: It would be nice to have a a test case in here that asserts that write timeout is overridden when profiling is enabled.